hs/WebParsing/TimeTableParser.hs

```diff
@@ -17,15 +18,111 @@ import WebParsing.TimeConverter
 import Control.Monad.IO.Class (liftIO, MonadIO)
 import Control.Monad.Trans.Reader (ReaderT)
 import Config (databasePath, timetableUrl)
+import Database.Tables (Department(Department))
 
 parseTT :: IO ()
 parseTT = do
     rsp <- simpleHTTP (getRequest timetableUrl)
     body <- getResponseBody rsp
+
     let depts = getDeptList $ parseTags body
+    let deptCode = getDeptCode $ parseTags body
+    let deptName = getDeptName $ parseTags body
 
     runSqlite databasePath $
         mapM_ getDeptTimetable depts
+    runSqlite databasePath $ 
+        mapM_ insertDepartment $ init $ constructAssociationList deptName deptCode
```

Comment:
`runSqlite databasePath $ ` doesn't need to be declared twice when performing two distinct actions on the database. This snippet opens up a connection to the database, does some stuff, and then closes it. An easier way to think about it is that it just runs whatever you feed into it. So, I'm pretty sure you can just get rid of line 34. Take a look at some other code in Courseography that does the same thing.

I made this mistake at first, too. Inserting all of the courses into the database took a lot longer when we would perform one `runSqlite` call for each course. Now, it runs like lightning.